### PR TITLE
chore: address low hanging security updates

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

npm audit fix
```
< 60 vulnerabilities (2 low, 5 moderate, 40 high, 13 critical)
---
> 51 vulnerabilities (2 low, 5 moderate, 33 high, 11 critical)
```

## Test Plan

- walk through the ui to see if anything's out of place

## Commentary (optional)

```
7a8
> node_modules/antd-theme-generator/node_modules/har-validator/node_modules/ajv
36,51d36
< ansi-regex  3.0.0 || 4.0.0 - 4.1.0
< Severity: high
< Inefficient Regular Expression Complexity in chalk/ansi-regex - https://github.com/advisories/GHSA-93q8-gq69-wqmw
< Inefficient Regular Expression Complexity in chalk/ansi-regex - https://github.com/advisories/GHSA-93q8-gq69-wqmw
< fix available via `npm audit fix`
< node_modules/@webpack-contrib/schema-utils/node_modules/ansi-regex
< node_modules/webpack-dev-server/node_modules/cliui/node_modules/ansi-regex
< node_modules/webpack-dev-server/node_modules/string-width/node_modules/ansi-regex
< node_modules/webpack-dev-server/node_modules/wrap-ansi/node_modules/ansi-regex
< 
< async  2.0.0 - 2.6.3
< Severity: high
< Prototype Pollution in async - https://github.com/advisories/GHSA-fwr7-v2mv-hh25
< fix available via `npm audit fix`
< node_modules/async
< 
70d54
< node_modules/source-map-explorer/node_modules/ejs
81,93d64
< eventsource  <1.1.1
< Severity: critical
< Exposure of Sensitive Information in eventsource - https://github.com/advisories/GHSA-6h5x-7c5m-7cr7
< fix available via `npm audit fix`
< node_modules/eventsource
< 
< follow-redirects  <=1.14.7
< Severity: high
< Exposure of Sensitive Information to an Unauthorized Actor in follow-redirects - https://github.com/advisories/GHSA-pw2r-vq6v-hr8c
< Exposure of sensitive information in follow-redirects - https://github.com/advisories/GHSA-74fj-2j2h-c42q
< fix available via `npm audit fix`
< node_modules/follow-redirects
< 
148d118
< node_modules/json-schema
152d121
<   node_modules/jsprim
159,165d127
< node_modules/minimist
< 
< moment  2.18.0 - 2.29.3
< Severity: high
< Inefficient Regular Expression Complexity in moment - https://github.com/advisories/GHSA-wc69-rhjr-hc9g
< fix available via `npm audit fix`
< node_modules/moment
243,250d204
< terser  <4.8.1 || >=5.0.0 <5.14.2
< Severity: high
< Terser insecure use of regular expressions before v4.8.1 and v5.14.2 leads to ReDoS - https://github.com/advisories/GHSA-4wf5-vphf-c2xc
< Terser insecure use of regular expressions before v4.8.1 and v5.14.2 leads to ReDoS - https://github.com/advisories/GHSA-4wf5-vphf-c2xc
< fix available via `npm audit fix`
< node_modules/terser
< node_modules/terser-webpack-plugin/node_modules/terser
< 
277,295c231
< trim-newlines  <3.0.1
< Severity: high
< Uncontrolled Resource Consumption in trim-newlines - https://github.com/advisories/GHSA-7p7h-4mm5-852v
< fix available via `npm audit fix`
< node_modules/trim-newlines
<   meow  3.4.0 - 5.0.0
<   Depends on vulnerable versions of trim-newlines
<   node_modules/meow
< 
< url-parse  <=1.5.8
< Severity: critical
< Incorrect hostname / protocol due to unstripped leading control characters. - https://github.com/advisories/GHSA-jf5r-8hm2-f872
< Authorization Bypass Through User-Controlled Key in url-parse - https://github.com/advisories/GHSA-hgjh-723h-mx2j
< Authorization bypass in url-parse - https://github.com/advisories/GHSA-rqff-837h-mm52
< Incorrect returned href via an '@' sign but no user info and hostname - https://github.com/advisories/GHSA-8v38-pw62-9cw2
< fix available via `npm audit fix`
< node_modules/url-parse
< 
< 60 vulnerabilities (2 low, 5 moderate, 40 high, 13 critical)
---
> 51 vulnerabilities (2 low, 5 moderate, 33 high, 11 critical)

```

the remaining ones seem dev only
```
hrz ➜ react git:(sec-update) npm ls antd-theme-generator
determined-webui@1.0.0 /home/hmd/projects/da/determined/webui/react
└─┬ antd-theme-webpack-plugin@1.3.9
  └── antd-theme-generator@1.2.11

hrz ➜ react git:(sec-update) npm ls react-scripts
determined-webui@1.0.0 /home/hmd/projects/da/determined/webui/react
├─┬ @craco/craco@6.4.0
│ └── react-scripts@4.0.3 deduped
├─┬ craco-less@1.20.0
│ └── react-scripts@4.0.3 deduped
├─┬ craco-sass-resources-loader@1.1.0
│ └── react-scripts@4.0.3 deduped invalid: "^2.1.3" from node_modules/craco-sass-resources-loader
├─┬ react-app-rewired@2.1.8
│ └── react-scripts@4.0.3 deduped invalid: "^2.1.3" from node_modules/craco-sass-resources-loader
├── react-scripts@4.0.3 invalid: "^2.1.3" from node_modules/craco-sass-resources-loader
└─┬ storybook-preset-craco@0.0.6
  ├─┬ @storybook/preset-create-react-app@3.2.0
  │ └── react-scripts@4.0.3 deduped invalid: "^2.1.3" from node_modules/craco-sass-resources-loader
  └── react-scripts@4.0.3 deduped invalid: "^2.1.3" from node_modules/craco-sass-resources-loader
```

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ClaireNeveu** [comment]
Curious what we pick up a `moment` dependency through, notoriously bad and huge library that is very obsolete with the new JS Date apis.

**Comment by hamidzr** [comment]
> Curious what we pick up a `moment` dependency through, notoriously bad and huge library that is very obsolete with the new JS Date apis.
```
determined-webui@1.0.0 determined/webui/react
└─┬ antd@4.21.4
  ├── moment@2.29.3
  └─┬ rc-picker@2.6.10
    └── moment@2.29.3 deduped

```
it'd would be hard to take this one out unless they do it.

**Comment by ClaireNeveu** [comment]
Seems to be tracked here: https://github.com/ant-design/ant-design/issues/26759. We could use a webpack plugin to swap out the dependency on our end. Probably worth doing since moment.js has now been officially deprecated for two years.

**Comment by hamidzr** [comment]
we might have used https://docs.npmjs.com/cli/v8/commands/npm-shrinkwrap  in the past to that for another package
+ https://github.com/hamidzr/determined/blob/f008dcb075ec27b0d4b1f8a5d80870873131163a/webui/react/package.json#L11  we might do the same? (context for that ticket)

**Comment by ClaireNeveu** [comment]
Ticket to track it is here: https://determinedai.atlassian.net/browse/DET-8244

**Comment by ClaireNeveu** [comment]
PR looks good to me once it's been tested, btw.

File: webui/react/package-lock.json [hunk 1]
```diff
@@ -22750,7 +24380,7 @@
     },
     "node_modules/omnibar": {
       "version": "2.3.5",
-      "resolved": "https://github.com/hamidzr/omnibar.git#688f7240d23e27c037f7fd49df92cf238474503c",
+      "resolved": "git+ssh://git@github.com/hamidzr/omnibar.git#688f7240d23e27c037f7fd49df92cf238474503c",
```
Comment (keita-determined) [inline webui/react/package-lock.json:24383]: should be fixed

File: webui/react/package-lock.json [hunk 2]
```diff
@@ -49384,7 +52145,7 @@
       "dev": true
     },
     "omnibar": {
-      "version": "https://github.com/hamidzr/omnibar.git#688f7240d23e27c037f7fd49df92cf238474503c",
+      "version": "git+ssh://git@github.com/hamidzr/omnibar.git#688f7240d23e27c037f7fd49df92cf238474503c",
```
Comment (keita-determined) [inline webui/react/package-lock.json:52148]: same here

**Review by keita-determined — COMMENTED** [review]
these should be fixed

File: webui/react/package-lock.json (hunk 1 above)
Comment (hamidzr) [reply webui/react/package-lock.json:24383]: ah it seems it was caught by ci as well as expected I was worried it wasn't https://circleci.com/gh/determined-ai/determined/869755?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link

**Review by keita-determined — APPROVED** [review]
LGTM but when I run `npm i`, I see some diff in package-lock. Could you double-check if package-lock is stable after the command besides `omnibar's` ssh:// thing?

**Comment by hamidzr** [comment]
I'm going to bring the branch up to date with target for good measure as well

**Comment by hamidzr** [comment]
thanks for the reviews

**Pull request merged**